encodings/fsst/src/compute/mod.rs

```diff
@@ -59,7 +59,10 @@ impl TakeFn<&FSSTArray> for FSSTEncoding {
             array.symbols().clone(),
             array.symbol_lengths().clone(),
             take(array.codes(), indices)?,
-            take(array.uncompressed_lengths(), indices)?,
+            fill_null(
+                &take(array.uncompressed_lengths(), indices)?,
+                Scalar::from(0).cast(indices.dtype())?,
```

Comment:
Slight preference for `Scalar::new(indices.dtype(), ScalarValue::from(0))` Mostly to avoid copies